nucypher/blockchain/eth/token.py

```diff
@@ -649,29 +648,42 @@ def pending(self) -> Dict[int, HexBytes]:
         return self.__pending.copy()
 
     def __commitments_tracker_is_consistent(self) -> bool:
-        worker_address = self.worker.worker_address
-        tx_count_pending = self.client.get_transaction_count(account=worker_address, pending=True)
-        tx_count_latest = self.client.get_transaction_count(account=worker_address, pending=False)
-        txs_in_mempool = tx_count_pending - tx_count_latest
+        tx_count_latest = self.client.get_transaction_count(account=self.worker.worker_address, pending=False)
+        tx_pending = self.__get_tx_pending()
+        txs_in_mempool = len(tx_pending) - tx_count_latest
 
         if len(self.__pending) == txs_in_mempool:
             return True  # OK!
 
         if txs_in_mempool > len(self.__pending):  # We're missing some pending TXs
             return False
-        else:  # TODO #2429: What to do when txs_in_mempool < len(self.__pending)? What does this imply?
-            return True
+
+        self.__num_untracked_commitments = len(set(tx_pending) - set(self.__pending))
+        # Mempool has limit on the number on TXs, and may drop them resulting in missing pending TXs
+        if self.__num_untracked_commitments != 0:
+            s = "s" if self.__num_untracked_commitments > 1 else ""
```

Comment:
I'm a little confused. We define `__get_tx_pending()` as:
```python
    def __get_tx_pending(self):
        pending_block = self.client.w3.eth.getBlock('pending', True)
        tx_pending = [
            tx.hash for tx in pending_block['transactions']
            if tx.hash in self.__pending.values()
        ]
        return tx_pending
```

So basically `tx_pending` is always a subset of `self.__pending`. Therefore, what am I missing that allows `self.__num_untracked_commitments`, which is the difference between the two, to ever be > 1?

Reply:
That's a mistake, it should actually be the other way around:
`self.__num_untracked_commitments = len(set(self.__pending)) - set(tx_pending)`.

The logic is that if the number of internally tracked pending transactions `self.__pending` is larger than the number of pending transactions in the block `tx_pending`, it would mean that one of the commitment transactions was dropped from the mempool.

Thanks for catching this!